Sources/TensorFlow/Operators/Basic.swift

```diff
@@ -1325,34 +1325,37 @@ extension Tensor.IndexPath {
 //===------------------------------------------------------------------------------------------===//
 
 extension Tensor {
-  /// Returns `true` if the given axis is in the range `[-rank, rank)`.
+  /// Returns `true` if `axis` is in `-rank..<rank`, indicating it refers to an axis of `self`.
   @usableFromInline
-  internal func isAxisInRange<T: BinaryInteger>(_ axis: T) -> Bool {
+  internal func isInRange<T: BinaryInteger>(axis: T) -> Bool {
     let axis = Int(axis)
     return axis >= -rank && axis < rank
   }
 
-  /// Returns `true` if the given scalar tensor is in the range `[-rank, rank)`.
+  /// Returns `true` iff all elements in `axis` are in `-rank..<rank`, indicating they refer to an
+  /// axis of `self`.
   @usableFromInline
-  internal func isAxisInRange(
-    _ axis: Tensor<Int32>,
+  internal func isInRange(
+    axis: Tensor<Int32>,
     file: StaticString = #file,
     line: UInt = #line
   ) -> Bool {
     precondition(axis.rank == 0, "Axis must have rank 0.", file: file, line: line)
-    return areAxesInRange(axis.scalars)
+    return isInRange(axes: axis.scalars)
   }
 
-  /// Returns `true` if all given axes are in the range `[-rank, rank)`.
+  /// Returns `true` iff all elements in `axis` are in `-rank..<rank`, indicating they refer to an
+  /// axis of `self`.
   @usableFromInline
-  internal func areAxesInRange<T: BinaryInteger>(_ axes: [T]) -> Bool {
-    return !axes.contains(where: { !isAxisInRange($0) })
+  internal func isInRange<T: BinaryInteger>(axes: [T]) -> Bool {
```

Comment:
Since you're taking the trouble to thread `#file` and `#line` through `isInRange(axis:)`, don't you want them here as well, forwarding on to the call in the body?

Reply:
Sorry for the noise, I obviously got lost in the tangle.